tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -441,6 +720,618 @@ parseDwarfInfo(const char *line)
+		Dwarf_Die parent = nestedClassesToPopulate[i].second;
+		Dwarf_Die nestedClass = NULL;
+		unordered_map<double, Dwarf_Die>::iterator entryToFind = createdDies.find(nestedClassesToPopulate[i].first);
+		if (createdDies.end() != entryToFind) {
+			nestedClass = entryToFind->second;
+			if (NULL != nestedClass) {
+				/* Get the last child of the die */
+				Dwarf_Die lastChild = parent->_child;
+				if (NULL != lastChild) {
+					while (NULL != lastChild->_sibling) {
+						lastChild = lastChild->_sibling;
+					}
+				}
+
+				/* Modify the linked list of DIEs to remove the nested class */
+				if (NULL != nestedClass->_previous) {
+					nestedClass->_previous->_sibling = nestedClass->_sibling;
+				}
+				if (NULL != nestedClass->_sibling) {
+					nestedClass->_sibling->_previous = nestedClass->_previous;
+				}
+				nestedClass->_sibling = NULL;
+				nestedClass->_previous = NULL;
+
+				/* Set the nested class DIE to be a child of the parent */
+				if (NULL == lastChild) {
+					parent->_child = nestedClass;
+				} else {
+					lastChild->_sibling = nestedClass;
+					nestedClass->_previous = lastChild;
+				}
+			}
+		}
+	}
+}
+
+/**
+ * Sets the name, size and declaring file attributes for a DIE that is passed in
+ * param[in] dieName: The name of the entity
+ * param[in] dieSize: The size of the entity (in bytes)
+ * param[out] currentDie: the DIE to populate with the attributes.
+ */
+static int
+setDieAttributes(const string dieName,
+	const unsigned int dieSize,
+	Dwarf_Die currentDie,
+	Dwarf_Error *error)
+{
+	int ret = DW_DLV_OK;
+	/* If the die already had attributes, update them */
+	if (NULL != currentDie->_attribute) {
+		Dwarf_Attribute tmp = currentDie->_attribute;
+		while (NULL != tmp) {
+			switch (tmp->_type) {
+				case DW_AT_name:
+					tmp->_stringdata = strdup(dieName.c_str());
+					break;
+				case DW_AT_byte_size:
+					tmp->_udata = dieSize;
+					break;
+				case DW_AT_decl_file:
+					tmp->_stringdata = strdup(Dwarf_CU_Context::_fileList.back().c_str());
+					break;
+			}
+			tmp = tmp->_nextAttr;
+		}
+	} else {
+		/* Create attributes for name, size, declaring file. */
+		Dwarf_Attribute name  = new Dwarf_Attribute_s();
+		Dwarf_Attribute size = new Dwarf_Attribute_s();
+		Dwarf_Attribute declFile = new Dwarf_Attribute_s();
+
+		if ((NULL == name) || (NULL == size) || (NULL == declFile)) {
+			ret = DW_DLV_ERROR;
+			setError(error, DW_DLE_MAF);
+		} else {
+			name->_type = DW_AT_name;
+			name->_nextAttr = size;
+			name->_form = DW_FORM_string;
+			name->_sdata = 0;
+			name->_udata = 0;
+			name->_stringdata = strdup(dieName.c_str());
+			name->_refdata = 0;
+			name->_ref = NULL;
+
+			size->_type = DW_AT_byte_size;
+			size->_nextAttr = declFile;
+			size->_form = DW_FORM_udata;
+			size->_sdata = 0;
+			size->_udata = dieSize;
+			size->_refdata = 0;
+			size->_ref = NULL;
+
+			declFile->_type = DW_AT_decl_file;
+			declFile->_nextAttr = NULL;
+			declFile->_form = DW_FORM_string;
```

Comment:
DW_AT_decl_file is actually to be of form DW_FORM_sdata, not string. The number is to be an index into the list of file names returned by dwarf_srcfiles() rather than the file name itself.